lua/null-ls/info.lua

```diff
@@ -40,65 +42,137 @@ local make_window = function(height_percentage, width_percentage)
+        return vim.tbl_map(function(method)
+            return methods.internal[method]:lower()
+        end, available_methods)
+    end
 
-    local log_path = c.get().debug and log:get_path()
-        or "not enabled (this is normal; see the README if you need to enable logging)"
-    table.insert(lines, "null-ls log: " .. log_path)
+    local get_supported_filestypes = function(source)
+        local filetypes = vim.tbl_keys(source.filetypes)
+        return vim.tbl_map(function(ft)
+            return ft == "_all" and "*" or ft
+        end, filetypes)
+    end
 
-    local ft = api.nvim_buf_get_option(bufnr, "filetype")
-    vim.list_extend(lines, { "Detected filetype: " .. ft, "" })
+    local create_source_info = function(source)
+        local info_lines = {
+            fmt("* name: %s", source.name),
+            fmt("* filetypes: %s", str_list(get_supported_filestypes(source))),
+            fmt("* methods: %s", str_list(get_methods_per_source(source))),
+            "",
+        }
+        return info_lines
+    end
+
+    local create_active_sources_info = function(ft)
+        local info_lines = {
+            "Active source(s)",
+        }
 
-    local active_sources = M.get_active_sources(bufnr, ft)
-    local source_count = 0
-    for _, sources in pairs(active_sources) do
-        source_count = source_count + #sources
+        for _, source in ipairs(sources.get_available(ft)) do
+            info_lines = vim.list_extend(info_lines, create_source_info(source))
+            table.insert(highlights, { "Title", "name:.*\\zs" .. source.name .. "\\ze" })
+        end
+        table.insert(highlights, { "Type", info_lines[1] })
+        return info_lines
     end
 
-    vim.list_extend(lines, { source_count .. " source(s) active for this buffer:", "" })
-    for method, sources in pairs(active_sources) do
-        table.insert(lines, methods.readable[method] .. ": " .. table.concat(sources, ", "))
+    local create_supported_methods_info = function(ft)
+        local supported_methods = sources.get_supported(ft)
+
+        local info_lines = {
+            "Supported source(s)",
+        }
+        -- the metadata is indexed by the builtin-names
+        for method, names in pairs(supported_methods) do
+            info_lines = vim.list_extend(info_lines, {
+                fmt("* %s: %s", method, str_list(names)),
+            })
+        end
+
+        table.insert(highlights, { "Type", info_lines[1] })
+        return info_lines
     end
 
-    local win_info = make_window(0.8, 0.7)
-    local win_bufnr, win_id = win_info.bufnr, win_info.win_id
+    local create_logging_info = function()
+        local info_lines = {
+            "Logging",
+            "* current level: " .. c.get().log.level,
+            "* path: " .. log:get_path(),
+        }
+        table.insert(highlights, { "Type", info_lines[1] })
+        return info_lines
+    end
+
+    local lines = {}
+
+    local header = {
+        "Null-LS",
```

Comment:
```suggestion
        "null-ls",
```

Really nitpicky, but...

Reply:
I actually wanted to center this, but couldn't figure out the minimal steps for that. There are also a few more tweaks I want to make, but they can be in a different PR.